nova/virt/vmwareapi/vmops.py

```diff
@@ -1535,6 +1531,22 @@ def _destroy_instance(self, context, instance, destroy_disks=True):
                 self._session._call_method(self._session.vim,
                                            "UnregisterVM", vm_ref)
                 LOG.debug("Unregistered the VM", instance=instance)
+                cluster_config = self._session._call_method(
+                    vutil, "get_object_property", self._cluster,
+                    "configurationEx")
+
+                # Check if the vm to unregister is the last vm in it's group
+                for group in cluster_config.group:
+                    for vm_group in vm_groups:
+                        if group.name == vm_group.name:
```

Comment:
If all you need are the names of the groups the VM is in, then you should create a set of names and just do a
```python
if group.name not in vm_group_names:
    continue
```